virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -1076,9 +1181,96 @@ def metamorphose_community(self) -> None:
         """Handle metamorphosis for all applicable cohorts in the community."""
 
         # Iterate over a static list of cohort values
-        for cohort in list(self.cohorts.values()):
+        for cohort in list(self.active_cohorts.values()):
             if (
                 cohort.functional_group.development_type == DevelopmentType.INDIRECT
                 and (cohort.mass_current >= cohort.functional_group.adult_mass)
             ):
                 self.metamorphose(cohort)
+
+    def update_migrated_and_aquatic(self, dt: timedelta64) -> None:
+        """Handles updating timing on frozen migrated and aquatic cohorts.
+
+        Args:
+            dt: The amount of time passed in the update (days).
+
+        """
+
+        dt_float = float(dt / timedelta64(1, "D"))
+        for cohort in list(self.migrated_cohorts.values()):
+            cohort.remaining_time_away -= dt_float
+            if cohort.remaining_time_away <= 0:
+                self.reintegrate_cohort(cohort, source="migrated")
+
+        for cohort in list(self.aquatic_cohorts.values()):
+            cohort.remaining_time_away -= dt_float
+            if cohort.remaining_time_away <= 0:
+                self.reintegrate_cohort(cohort, source="aquatic")
```

Comment:
So this function does have the consequence that `remaining_time_away` might be left as less than zero afterwards. Not sure if you want to just set it to zero in this case? It might be nice if other code can rely on it being >=0 (not sure if other code is actually using it though)